aiidalab/widgets.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+"""AiiDA lab basic widgets."""
+
+from threading import Thread
+from time import sleep, time
+
+import traitlets
+import ipywidgets as ipw
+
+
+class _StatusWidgetMixin:
+    """Show temporary messages for example for status updates."""
+
+    def __init__(self, *args, **kwargs):
+        self._clear_timer = 0
+        super().__init__(*args, **kwargs)
+
+    def _clear_value_after_delay(self, delay):
+        self._clear_timer = time() + delay  # reset timer
+        sleep(delay)
+        if time() > self._clear_timer:
```

Comment:
this check appears redundant to me

Reply:
The timer can be reset by additional calls, so it is not redundant.

Reply:
Let me iterate one more time on this. There seems to be a logical problem here.

It appears to me that you want to leave a back-side door to prevent cleaning of the printed value after `_clear_value_after_delay` has been called, right?

There are two questions in this respect:
1. Do we really need such a possibility? And why?
1. If this possibility is really needed, can we add a function that is called `stop_cleaning` that will act according to its name? The current implementation appears quite confusing to me.

```
class _StatusWidgetMixin:
    """Show temporary messages for example for status updates."""

    def __init__(self, *args, **kwargs):
        self._stop_cleaning = False
        super().__init__(*args, **kwargs)

    def _clear_value_after_delay(self, delay):
        sleep(delay)
        if self._stop_cleaning:  # Emergency stop.
            self._stop_cleaning = False
        self.value = ''
    
    def stop_cleaning(self):
        self._stop_cleaning = True

    def show_temporary_message(self, value, clear_after=3):
        self.value = value
        if clear_after > 0:
            Thread(target=self._clear_value_after_delay, args=(clear_after,)).start()
```

Reply:
I'm 99% sure that there was no logical issue, but I've improved the clarify of the code with 9decb563094a64f03589c9dacdb02d6aa627ccd3 .

Reply:
ok